src/plugins/intel_cpu/src/nodes/interpolate.cpp

```diff
@@ -1763,27 +1762,21 @@ class InterpolateShapeInferFactory : public ShapeInferFactory {
 public:
     InterpolateShapeInferFactory(std::shared_ptr<ov::Node> op) : m_op(op) {}
     ShapeInferPtr makeShapeInfer() const override {
-        IShapeInfer::port_mask_t port_mask = 0x00;
         if (auto interp4 = ov::as_type_ptr<ov::opset4::Interpolate>(m_op)) {
             const auto &attr = interp4->get_attrs();
-
-            if (attr.shape_calculation_mode == ngInterpShapeCalcMode::SCALES) {
-                port_mask = PortMask(Interpolate::SCALES_ID, Interpolate::AXES_ID);
-            } else if (attr.shape_calculation_mode == ngInterpShapeCalcMode::SIZES) {
-                port_mask = PortMask(Interpolate::TARGET_SHAPE_ID, Interpolate::AXES_ID);
-            } else {
-                OPENVINO_ASSERT(false, "Unsupported interpolate shape calculation mode");
-            }
+            const auto is_supported_mode = (attr.shape_calculation_mode == ngInterpShapeCalcMode::SCALES) ||
+                                           (attr.shape_calculation_mode == ngInterpShapeCalcMode::SIZES);
+            OPENVINO_ASSERT(is_supported_mode, "Unsupported interpolate shape calculation mode");
+            return make_shape_inference(m_op);
```

Comment:
I'm not sure if custom factory is required as is CPU, this is reason why left to avoid regressions.
I follow your suggestion if it can be safely removed. I Can do it as separate PR to not block working part.